### PR TITLE
fix-bug-nlg-metaevaluation

### PR DESCRIPTION
**Pull request by pfliu-nlp**
There are some bugs in the meta-evaluation task, and the calculated overall results are wrong. After some advice from @qinyiwei , this PR is finished:
* fix bugs
* add strict test

### CODE REVIEW

File: explainaboard/metrics/nlg_meta_evaluation.py [hunk 1]
```diff
@@ -188,10 +188,21 @@ def _count(self, score: list, config: Optional[MetricConfig] = None):
...
+                elif (
+                    score[i][0] - score[j][0] >= config.threshold
+                ):  # system i is better than system j
+                    if score[i][1] > score[j][1]:
+                        conc += 1
+                    else:
+                        disc += 1
+                    num += 1
+                else:  # system i is worse than system j
+                    if score[i][1] < score[j][1]:
```
Comment (odashi) [inline explainaboard/metrics/nlg_meta_evaluation.py:205]: It looks the same stats are used multiple times. Could you apply following variables to judge the conditions?

- `manual_diff = score[i][0] - score[j][0]`
- `system_diff = score[i][1] - score[j][1]`

File: explainaboard/metrics/nlg_meta_evaluation.py [hunk 2]
```diff
@@ -234,19 +245,27 @@ class PearsonCorrelation(CorrelationMetric):
...
+        elif config.group_by == "system":
+            for _, group_vals in scores.items():
+                manual_scores = [val[0] for val in group_vals]
+                system_scores = [val[1] for val in group_vals]
+                if len(manual_scores) == 0 or len(system_scores) == 0:
+                    continue
+                manual_score.append(sum(manual_scores) / len(manual_scores))
+                system_score.append(sum(system_scores) / len(system_scores))
+        else:
+            raise NotImplementedError
```
Comment (odashi) [inline explainaboard/metrics/nlg_meta_evaluation.py:267]: ValueError with an appropriate message is better.

File: integration_tests/nlg_meta_eval_test.py [hunk 3]
```diff
@@ -36,6 +36,6 @@ def test_da_cs_en(self):
         processor = get_processor_class(TaskType.nlg_meta_evaluation)()
 
         sys_info = processor.process(metadata, data)
-
-        self.assertGreater(len(sys_info.results.analyses), 0)
-        self.assertGreater(len(sys_info.results.overall), 0)
+        self.assertAlmostEqual(
```
Comment (odashi) [inline integration_tests/nlg_meta_eval_test.py:39]: I think the original assertion does some different checks, and needs to be maintained as well.

File: explainaboard/metrics/nlg_meta_evaluation.py (hunk 1 above)
Comment (pfliu-nlp) [reply explainaboard/metrics/nlg_meta_evaluation.py:205]: good suggestion

File: explainaboard/metrics/nlg_meta_evaluation.py [hunk 4]
```diff
@@ -188,21 +188,20 @@ def _count(self, score: list, config: Optional[MetricConfig] = None):
         num = 0
         for i in range(1, len(score)):
             for j in range(0, i):
-                if (
-                    abs(score[i][0] - score[j][0]) < config.threshold
-                    or abs(score[i][0] - score[j][0]) == 0
-                ):
+                manual_diff = score[i][0] - score[j][0]
+                system_diff = score[i][1] - score[j][1]
+                if abs(manual_diff) < config.threshold or abs(manual_diff) == 0:
```
Comment (odashi) [inline explainaboard/metrics/nlg_meta_evaluation.py:193]: I noticed that the following is much simpler:
```python
if manual_diff >= config.threshold:
   ...  # i is better than j
elif manual_diff <= -config.threshold:
   ...  # i is worse than j
#that's all
```

**Review by odashi — APPROVED** [review]
Approves it, but there is one comment

**Review by neubig — APPROVED** [review]

File: explainaboard/metrics/nlg_meta_evaluation.py (hunk 4 above)
Comment (pfliu-nlp) [reply explainaboard/metrics/nlg_meta_evaluation.py]: Thanks. Fixed.

**Pull request merged**